### PR TITLE
fix(scheduling): adopt a manually-armed panel mid-window (fixes #212)

### LINKED ISSUE #212: Skipped arm strands the panel: no disarm is ever scheduled when the panel is armed_away at arm time
## What happened

Schedule "Night arming" (all 7 weekdays, arm `23:00` → disarm `06:00`, enabled) armed the panel every night as expected, but on the morning of 2026-08-22 the panel was still `armed_home` when the household woke up. It was disarmed by hand at 07:25:50.

Stored record afterwards:

```json
"last_skip_reason": "away_active",
"last_armed_at":    "2026-08-21T06:00:00.407422+00:00",   // = Aug 20 23:00 local
"last_disarmed_at": "2026-08-22T06:00:00.094920+00:00"    // = Aug 21 23:00 local
```

`/config/.storage/abode_security_schedules` has an mtime of exactly `2026-08-21 23:00:00 -0700` — the scheduler wrote nothing at 06:00, because nothing was scheduled to run.

## Why

The arm and disarm sides are asymmetric by design. `_register_pair_timers` registers only the recurring daily arm callback:

> Disarm handles are NOT registered here — they're created on-demand by a successful arm (or by reconciliation).

`async_arm` creates that one-shot disarm on exactly two paths: a successful arm from `disarmed`, and the `already_home` branch where the panel is already in the mode the schedule would have set.

At 23:00 on 2026-08-21 the panel was `armed_away` (an unrelated presence bug had triggered vacation mode at 22:32). `async_arm` took the `armed_away` branch, which records the skip and **returns before `_schedule_disarm`**. The panel was then armed to Home manually at 23:37 — outside the scheduler, so still no disarm timer. Nothing re-evaluated between then and morning, so no 06:00 disarm existed.

## The arm-time rule is correct — don't change it

It is tempting to schedule the disarm on the `away_active` path too. That would be a security regression: on a genuine trip it plants a timer that disarms an empty house at 06:00. Refusing to pre-commit to a disarm while the panel is in Away is the right call, and so is the existing rule — **schedule the disarm only on a successful arm, or when the panel is already in the mode the schedule would have armed**.

The defect is not at arm time. It is that the scheduler evaluates panel state only at the arm instant and at startup, so a panel that *enters* the target mode later in the window is never adopted.

## Proposed fix

Make the manual-override listener bidirectional. `_on_panel_state_changed` currently returns early when `new_state.state == "armed_home"`, so it only reacts to transitions *leaving* Home (cancelling a pending disarm). Add the mirror case: a transition *into* `armed_home` from a non-scheduler context, while inside an enabled pair's arm→disarm window and with no pending disarm handle, should adopt the panel and register the one-shot disarm.

The logic already exists — startup reconciliation handles precisely this state ("in window, still Home" → `async_call_later` the disarm). It just only runs at startup. Extracting that into a helper shared by reconciliation and the listener should be most of the work.

Edge cases worth covering in tests:

- Manual arm to Home inside the window → disarm registered for the pair's `disarm_time`.
- Manual arm to Home *outside* any window → nothing registered.
- Manual arm to Home when a disarm handle already exists → no double registration.
- Scheduler-driven transitions still ignored via the `CONTEXT_ID_PREFIX` check.
- Manual arm to Home inside the window, then manually disarmed before `disarm_time` → the existing leave-Home path cancels cleanly.
- Two enabled pairs with overlapping windows → each gets at most one handle.

## Impact

Any Away at 23:00 strands the panel with no morning release, not just this one caused by a presence bug. A real trip that ends overnight hits the same path.

https://claude.ai/code/session_018Yp2hwnqsDW2K9jg3TcEHN

### PR DESCRIPTION
**Pull request by molant**
Fixes #212.

## The bug

When the panel was already `armed_away` at arm time, the scheduler skipped the arm edge and never scheduled a disarm. An Away at 23:00 left the panel armed for the whole of the next day — the pair's window closed with nothing to release it.

## The fix

The panel listener becomes bidirectional. It already gave the panel *up* when it left Home (`_handle_manual_override`); it now takes it *on* when it arrives. `_handle_manual_arm` runs on the edge into `armed_home`, walks the pairs whose window contains now, stamps the runtime record and installs the disarm timer.

The arm-time rule is deliberately unchanged. Relaxing it — pre-committing to a disarm while the panel is in Away — would plant a timer that unarms an empty house on a genuine trip. Adopting the panel once it is actually Home is the safe half of the same idea.

### Guards

Six conditions keep adoption from ever scheduling a disarm nobody asked for (a seventh, the edge-not-level dispatch, lives in the caller). Two of them are worth calling out: the loop-top panel read and the pre-timer one both test for `armed_away`/`disarmed` **by name**, not for "not Home". An availability blip is not evidence the panel left, and declining on one would strand the pair with no handle and no later edge to re-adopt it — the `#216` guard in `_on_panel_state_changed` returns on a blip before dispatching the override and before updating `_last_panel_state`, so recovery is not an edge. That is #212 again by a narrower route.

The anchor and the skip reason are written in separate store calls with the ownership re-check between them, so a declined adoption cannot leave `already_home` on a pair it never adopted (which would also clobber the `manual_override` the race winner had just written).

### Also in here

- **`_clear_spent_disarm_handle`** — a fired one-shot disarm used to leave its `_pending_handles` entry behind; nothing removed it. The dict had drifted from meaning "a disarm is pending" to "a disarm was scheduled at some point since the last reload", and a stale key swallowed the adoption on the night after any completed disarm. Independently, this removes a path where `async_update`'s `had_pending_disarm` could re-register a `delay ≈ 0` timer concurrent with an in-flight `_disarm_impl`.
- **`in_arm_window`** in the state machine — the mirror of `expected_disarm_at`. Anchored on the most recent `arm_time` occurrence and matching *that occurrence's* weekday rather than `now`'s, so overnight pairs answer correctly (Tuesday 02:00 is inside the Monday window). Pins `fold=0` and compares absolute instants, so fall-back nights don't report a window the panel is inside as closed.

## Accepted trade-offs

Recorded in `docs/ARCHITECTURE.md` rather than fixed here:

- A flapping mode source now costs three store writes and two `schedule_skipped` events per Home→Away→Home cycle where it used to cost one write. The obvious guard (refuse adoption after a same-window `manual_override`) was rejected because it cannot distinguish a flap from the legitimate disarm-at-01:00 / re-arm-at-01:30 flow.
- A sub-hour spring-forward seam in `in_arm_window`, for an `arm_time` inside the skipped hour.
- A user arming Home *during* a scheduled arm's confirmation poll now reports both `schedule_skipped(already_home)` and `schedule_fired`. Cosmetic — one disarm survives, at the right instant.

## Testing

`./scripts/check.sh` green end to end: ruff, mypy, pyright (35 source files), **923 tests**. `state_machine.py` at 100% coverage.

New: `TestManualArmAdoption` (25 cases — the reported failure end to end, both window boundaries, overnight weekday matching, disabled pairs, self-context suppression, mid-pass panel departure parametrized over both leave modes, availability blips at each write, the flap, a full two-night sequence) and `TestInArmWindow` (boundaries, all-weekday, Madrid fall-back and spring-forward).

Both new guards and the write split were mutation-tested — each has a test that fails when the guard is removed or broadened.

https://claude.ai/code/session_0121PY3FMBcNeeaq58p7DRoj

### CODE REVIEW

File: custom_components/abode_security/scheduling/manager.py [hunk 1]
```diff
@@ -1261,13 +1295,225 @@ def _on_panel_state_changed(self, event: Event[EventStateChangedData]) -> None:
...
+            # widen a security-relevant path on an inference rather than an
+            # observation.  The residual is a real one and is recorded in
+            # ARCHITECTURE.md next to the sibling dropout gap: a reload that lands
+            # while the panel is unavailable, on a night whose arm was skipped,
+            # strands the panel for the rest of the window.  It fails in the safe
+            # direction — armed, never unarmed — and
+            # `test_first_ever_panel_reading_is_not_a_transition` pins it so
+            # closing it later is a deliberate change.
+            if previous is not None and previous != "armed_home":
+                self._track(self._handle_manual_arm())
```
[the remark this answers is not in the document]
Comment (molant) [reply custom_components/abode_security/scheduling/manager.py:1349]: Addressed in 7596d33 — this was real, and reproducible.

I confirmed it before fixing. With two in-window pairs and the override's first store write stalled, the observed order was `ovr:start → arm:start → arm:end → ovr:end`, and the second pair ended with no handle, `last_skip_reason=manual_override` and `last_armed_at == last_disarmed_at` (IDLE) while the panel was `armed_home` and in-window — a stranded pair, which is #212's own symptom one pair over from the fix for it.

(One note in case it helps anyone re-running this: my first repro attempt was invalid. `FakeClock` is frozen, so both handlers stamp the same instant and `derive_state`'s strict `last_armed_at > last_disarmed_at` declines the adoption for a harness reason rather than a race. The control run with no stall showed the identical "stranded" result, which is what gave it away — a real race heals there. Advancing the clock between the two handlers isolates the actual bug.)

Fix is the shared lock you suggested: `_edge_lock`, an `asyncio.Lock` held by both `_handle_manual_arm` and `_handle_manual_override` for their whole pass. Acquisition is FIFO and the tasks are created in event-bus order, so it is event order rather than merely one-at-a-time. Each handler also samples its own `now` *inside* the lock, so a queued one stamps an instant strictly later than the pass it waited on — hoisting that sample would silently restore the bug, and there is a comment saying so.

Pinned by `test_an_override_cannot_interleave_with_a_later_adoption`; replacing the lock with a `nullcontext` fails exactly that test.

Two things fell out that were worth more than the fix itself:

- The lock made `_handle_manual_override` unable to reach the post-write `derive_state` guard in `_handle_manual_arm`, which dropped that branch out of test coverage entirely while leaving a comment that still explained it in terms of the override. The guard is still live for unlocked writers, so the docs now name the real cast — `last_disarmed_at` writers (`_arm_impl`'s `away_active`/`panel_unavailable`, `_disarm_impl`, reconciliation) and the `last_armed_at`-regressing one (`_arm_impl`'s success path, i.e. the confirmation-poll overlap) — and `test_the_post_write_guard_declines_when_another_writer_releases_the_pair` reaches it again.
- Reconciliation is deliberately *not* under the lock. `ARCHITECTURE.md` records why that gap is narrower than it looks (no await between reconcile's re-read, its `expected_disarm_at` and `_set_disarm_handle`), and what the reachable residue actually is (a stale `reconcile_panel_not_home` reason on a pair that keeps its adopted timer).

Gates green: ruff, mypy/pyright, 925 tests.
